workspaces/src/result.rs

```diff
@@ -68,45 +65,54 @@ impl CallExecutionDetails {
     /// result. This conversion can fail if the structure of the internal state does
     /// not meet up with [`borsh::BorshDeserialize`]'s requirements.
-    pub fn borsh<T: borsh::BorshDeserialize>(&self) -> anyhow::Result<T> {
+    pub fn borsh<T: borsh::BorshDeserialize>(&self) -> Result<T> {
         let buf = self.raw_bytes()?;
-        borsh::BorshDeserialize::try_from_slice(&buf).map_err(Into::into)
+        borsh::BorshDeserialize::try_from_slice(&buf)
+            .map_err(BytesError::BorshError)
+            .map_err(Into::into)
     }
 
     /// Grab the underlying raw bytes returned from calling into a contract's function.
     /// If we want to deserialize these bytes into a rust datatype, use [`CallExecutionDetails::json`]
     /// or [`CallExecutionDetails::borsh`] instead.
-    pub fn raw_bytes(&self) -> anyhow::Result<Vec<u8>> {
-        let result: &str = match self.status {
-            FinalExecutionStatus::SuccessValue(ref val) => val,
-            FinalExecutionStatus::Failure(ref err) => anyhow::bail!(err.clone()),
-            FinalExecutionStatus::NotStarted => anyhow::bail!("Transaction not started."),
-            FinalExecutionStatus::Started => anyhow::bail!("Transaction still being processed."),
-        };
-        base64::decode(result).map_err(Into::into)
+    pub fn raw_bytes(&self) -> Result<Vec<u8>> {
+        let result = self.try_into_success_value()?;
+        base64::decode(result)
+            .map_err(BytesError::DecodeBase64Error)
+            .map_err(Into::into)
+    }
+
+    fn try_into_success_value(&self) -> Result<&str> {
+        match self.status {
+            FinalExecutionStatus::SuccessValue(ref val) => Ok(val),
+            FinalExecutionStatus::Failure(ref err) => Err(Error::ExecutionError(err.to_string())),
+            FinalExecutionStatus::NotStarted => {
+                Err(Error::ExecutionError("Transaction not started.".into()))
+            }
+            FinalExecutionStatus::Started => Err(Error::ExecutionError(
+                "Transaction still being processed.".into(),
+            )),
+        }
     }
 
     /// Convert the execution details into a Result if its status is not a successful one.
     /// Useful for checking if the call was successful and forwarding the error upwards.
-    fn try_into_result(self) -> anyhow::Result<Self> {
-        match self.status {
-            FinalExecutionStatus::Failure(ref err) => anyhow::bail!(err.clone()),
-            FinalExecutionStatus::NotStarted => anyhow::bail!("Transaction not started."),
-            FinalExecutionStatus::Started => anyhow::bail!("Transaction still being processed."),
-            _ => (),
-        };
+    fn try_into_result(self) -> Result<Self> {
+        self.try_into_success_value()?;
         Ok(self)
     }
 
-    pub(crate) fn from_outcome(outcome: FinalExecutionOutcomeView) -> anyhow::Result<Self> {
+    pub(crate) fn from_outcome(outcome: FinalExecutionOutcomeView) -> crate::result::Result<Self> {
```

Comment:
Unnecessary full path here too. There seem to be a few of those around, not sure if that was intentional, lmk your reasoning if it is